pkg/k8sutils/templates/pod/mount.go

```diff
@@ -10,9 +10,23 @@ import (
 	"github.com/StarRocks/starrocks-kubernetes-operator/pkg/common/hash"
 )
 
-func IsSpecialStorageClass(storageClassName *string) bool {
-	return storageClassName != nil && (common.EqualsIgnoreCase(*storageClassName, v1.EmptyDir) ||
-		common.EqualsIgnoreCase(*storageClassName, v1.HostPath))
+// SpecialStorageClassName returns the special storage class name of the storage volume, else return "".
+// Now we support HostPath and EmptyDir as special storage class.
+func SpecialStorageClassName(sv v1.StorageVolume) string {
+	if sv.HostPath != nil {
```

Comment:
what if the the sv.HostPath is provided and the sv.StorageClassName is provided as well and they mismatch?

Reply:
Good catch. I think the storage class name will have a high priority.